### ISSUE TITLE
how to get PiecewiseLinear.step()?

### ISSUE THREAD

**Issue by anonlim** [pytorch/ignite#3006]
## ❓ Questions/Help/Support

In  setting lr_scheduler, [Transformers for Text Classification with IMDb Reviews](https://pytorch-ignite.ai/tutorials/beginner/02-transformers-text-classification/) Tutorial

```python
lr_scheduler = PiecewiseLinear(
        optimizer, param_name="lr", milestones_values=milestones_values
    )
for batch in train_dataloader:
```

```python
    lr_scheduler.step()
```

i think there's no step() function in PiecewiseLinear( also, class ParamScheduler, BaseParamScheduler)

**Comment by sadra-barikbin** [comment]
@anonlim , thank you for your question. The code that actually gets run, is `train_step` function just below the snippet you mentioned. That snippet just shows the operations that should be done. Maybe we'd better change the statement above the snippet.

**Comment by vfdev-5** [comment]
> i think there's no step() function in PiecewiseLinear( also, class ParamScheduler, BaseParamScheduler)

That's right, there is no `.step()` method in `ParamScheduler`. We were thinking with @theory-in-progress while working on [code-generator](https://code-generator.pytorch-ignite.ai/) to add `lr_scheduler.step()` in addition to `lr_scheduler(engine)`.

**Comment by anonlim** [comment]
@sadra-barikbin @vfdev-5  thx for your help!!

@sadra-barikbin you mean,,, piecewiselinear func can be replaced by piecewiselinear function?

**Comment by sadra-barikbin** [comment]
Not really. I meant the code that actually gets run at each step is
```python
def train_step(engine, batch):  
    model.train()
    
    batch = {k: v.to(device) for k, v in batch.items()}
    outputs = model(**batch)
    loss = outputs.loss
    loss.backward()

    optimizer.step()
    optimizer.zero_grad()

    return loss
```
But where's the code related to lr scheduler? LR scheduler is attached to engine as follows and gets called at each iteration:
```python
trainer.add_event_handler(Events.ITERATION_STARTED, lr_scheduler)
```
For your information, Ignite's LR schedulers, when get called (LR scheduler is an object whose `__call__` method is run when one calls it), update optimizer's LR explicitly, i.e. they set it. So there's no `lr_scheduler.step()`

**Comment by anonlim** [comment]
@sadra-barikbin Oh, I fully understood. thank you for your help!!
